content_api/index/common_disk.py

```diff
@@ -0,0 +1,192 @@
+                except Exception:
+                    log.exception(f"Failed to load entry {folder_name}/{unique_id}. Skipping.")
+                    continue
+
+                # Return of None means package is blacklisted
+                if package is None:
+                    add_to_blacklist(content_type, unique_id)
+                    continue
+
+                index_package(package)
+
+                for version in package["versions"]:
+                    version_dep_check.append(version)
+
+                # Scenarios and heightmap share their index in the OpenTTD client.
+                # Their unique-id is also assigned by us (and not the user as with
+                # all the other content). As such, find the highest currently used
+                # number, and store that.
+                if content_type in (ContentType.SCENARIO, ContentType.HEIGHTMAP):
+                    set_if_higher_scenario_heightmap_id(int(unique_id, 16))
+
+            log.info("Loaded %d entries for %s", get_indexed_count(content_type), content_type.value)
+
+        log.info("Highest unique-id used by scenario/heightmap is %d", get_highest_scenario_heightmap_id())
+
+        set_dependency_check(True)
+
+        # Validate all loaded versions, this time with dependency checking on.
+        # This to make sure we are in a consistent state.
+        for version in version_dep_check:
+            errors = VersionMinimized().validate(VersionMinimized().dump(version))
+            if errors:
+                raise Exception("Failed to load content entries: %r" % errors)
+
+    def store_package(self, package):
+        os.makedirs(f"BaNaNaS/{package['content_type'].value}/{package['unique_id']}", exist_ok=True)
+        os.makedirs(f"BaNaNaS/{package['content_type'].value}/{package['unique_id']}/versions", exist_ok=True)
+
+        data = Global().dump(package)
+        with open(f"BaNaNaS/{package['content_type'].value}/{package['unique_id']}/global.yaml", "w") as fp:
+            fp.write(yaml_dump(data))
+
+        data = Authors().dump({"authors": package["authors"]})
+        with open(f"BaNaNaS/{package['content_type'].value}/{package['unique_id']}/authors.yaml", "w") as fp:
+            fp.write(yaml_dump(data))
+
+        for version in package["versions"]:
+            data = VersionMinimized().dump(version)
+            data["upload-date"] = date_string(data["upload-date"].replace("+00:00", "Z"))
+            upload_date = data["upload-date"].replace("-", "").replace(":", "")
+
+            # Make sure the overwrite fields are at the bottom; this just reads a
+            # bit easier.
+            data_overwrite = OrderedDict()
+            for field in Global().fields:
+                if field in data:
+                    data_overwrite[field] = data[field]
+                    del data[field]
+
+            with open(
+                f"BaNaNaS/{package['content_type'].value}/{package['unique_id']}/versions/{upload_date}.yaml", "w"
```

Comment:
i appreciate your dedication to getting all the capitals in the correct place :D